hrm-domain/scheduled-tasks/conversation-cleanup/index.ts

```diff
@@ -0,0 +1,234 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import { AccountSID } from '@tech-matters/types';
+import { getClient } from '@tech-matters/twilio-client';
+import {
+  getCachedParameters,
+  getSsmParameter,
+  loadSsmCache,
+  SsmParameterNotFound,
+} from '@tech-matters/ssm-cache';
+import subDays from 'date-fns/subDays';
+
+const MAX_CLEANUP_JOB_RETENTION_DAYS = 3650;
+
+const MAX_ACTIVE_CONVERSATION_HOURS = 24;
+
+const hrmEnv = process.env.NODE_ENV;
+
+/**
+ * Get the number of days to retain cleanup jobs for a given account
+ * @param accountSid
+ * @returns number of days to retain cleanup jobs
+ */
+const getCleanupRetentionDays = async (accountSid): Promise<number | undefined> => {
+  let ssmRetentionDays: number;
+  try {
+    ssmRetentionDays =
+      parseInt(
+        await getSsmParameter(
+          `/${process.env.NODE_ENV}/hrm/${accountSid}/transcript_retention_days`,
+        ),
+      ) || MAX_CLEANUP_JOB_RETENTION_DAYS;
+    console.debug(
+      `SSM parameter for transcript retention days set to ${ssmRetentionDays} for account ${accountSid}, so using that`,
+    );
+  } catch (err) {
+    if ((err as any) instanceof SsmParameterNotFound) {
+      console.debug(
+        `SSM parameter for transcript retention days not set for account ${accountSid}, so using default ${MAX_CLEANUP_JOB_RETENTION_DAYS}`,
+      );
+    } else {
+      console.error(
+        `Error trying to fetch /${process.env.NODE_ENV}/hrm/${accountSid}/transcript_retention_days ${err}, using default`,
+        err,
+      );
+    }
+    ssmRetentionDays = MAX_CLEANUP_JOB_RETENTION_DAYS;
+  }
+
+  // For now we are limiting the retention days to 365 days for all jobs and allowing for a
+  // global override on a per account basis. This may need to epand to be more granular in the
+  // future.
+  return Math.min(MAX_CLEANUP_JOB_RETENTION_DAYS, ssmRetentionDays);
+};
+
+const PAGE_SIZE = 40;
+
+type ConversationListResponse = {
+  conversations: { sid: string }[];
+};
+
+const removeConversationsBySids = async (
+  accountSid: AccountSID,
+  conversationSids: string[],
+  attemptToClose: boolean,
+) => {
+  const twilioClient = await getClient({ accountSid });
+  // eslint-disable-next-line @typescript-eslint/no-loop-func
+  const removePromises = conversationSids.map(async sid => {
+    if (attemptToClose) {
+      console.debug(`Closing conversation ${sid} from account ${accountSid}`);
+      try {
+        const result = await twilioClient.conversations.v1.conversations
+          .get(sid)
+          .update({ state: 'closed' });
+        if (result.state === 'closed') {
+          console.debug(`Closed conversation ${sid} from account ${accountSid}`);
+          return;
```

Comment:
Wouldn't be better to log this only if the state was `active` (or `open`) before the update? We are guaranteed it will be `closed` if code reaches this if. Unless only `active` channel are passed to this function?